### PR TITLE
fix: added more unit test cases

### PR DESCRIPTION
**Pull request by nikhilsaikethe**
### **PR Type**
Tests, Enhancement

___

### **Description**
- Add extensive LogStream component tests

- Introduce service unit tests coverage

- Improve test mocks and handlers

- Fix brittle UI tests and stubs

___

### Diagram Walkthrough

```mermaid
flowchart LR
  LogStreamSpec["LogStream.spec.ts"] -- adds comprehensive component tests --> VueApp["Vue components"]
  ServiceSpecs["tickets/billings/organizations.spec.ts"] -- mock http and assert endpoints --> HTTP["HTTP layer"]
  UtilsSpec["convertDataIntoUnitValue.spec.ts"] -- validate formatting and conversions --> DashboardUtils["Dashboard utils"]
  TestInfra["handlers.ts + stubs"] -- stabilize mocks --> TestRunner["Vitest env"]
  UIFixes["User/AddUser/Reports specs"] -- adjust stubs and expectations --> UIComponents["UI components"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Tests</strong></td><td><details><summary>14 files</summary><table>
<tr>
  <td><strong>LogStream.spec.ts</strong><dd><code>Rewrite and greatly expand LogStream tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8025/files#diff-5d1ec597d0896fc4768515852054db8c919e048d2e8f6013bfbc4c42fa7503da">+1197/-82</a></td>

</tr>

<tr>
  <td><strong>tickets.spec.ts</strong><dd><code>Add full CRUD tests for tickets service</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8025/files#diff-6326b5fdbd0493710b156bfb06237d5906fc493de034e872a879b9d59245d483">+565/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>billings.spec.ts</strong><dd><code>Add billing workflows and error tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8025/files#diff-8152eb5ae97615b253cd61eedea9147bf255cf444eefea25e1a4e954ffc7ee65">+547/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>convertDataIntoUnitValue.spec.ts</strong><dd><code>Test data unit conversion and formatting utils</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8025/files#diff-e9d8a984ed78af13e4d4424c605a3a55fe6ad2a3f7e403fad156ce17ffdc3bdd">+452/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>organizations.spec.ts</strong><dd><code>Cover organizations service endpoints and flows</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8025/files#diff-d891fff7a737f937ff3cad0710f07a99479d7160e31928259d1aec4ebfaf367d">+519/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>AddAlert.spec.ts</strong><dd><code>Add missing composable/service mocks for alerts</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8025/files#diff-84362398fea8abd9acf30f4f40ba8d4cf27348677454017bd4e6f568d5176ac6">+47/-0</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>User.spec.js</strong><dd><code>Stabilize user tests and adjust stubs</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8025/files#diff-f792c36cfff4d60265030596a1316a3fe207a1bcafc53cd0223a4873f9daf25b">+8/-20</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>users.spec.ts</strong><dd><code>Add users service tests (CRUD, roles)</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8025/files#diff-1e0367915c28073772fd91455c185fa522f17453fc868207b030f606fb7c0dee">+425/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>common.spec.ts</strong><dd><code>Add common utilities unit tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8025/files#diff-37bf940c4e06c75c7443e4342a8e2896adbb8cff23fad8f7227e4cd58537e6fb">+395/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>useStreams.spec.ts</strong><dd><code>Add tests for useStreams composable pagination</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8025/files#diff-513e2d53c46c553904a054e9c2cd12d33e800afc54fe5b4192c9782382ecd79b">+285/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>auth.spec.ts</strong><dd><code>Add authentication service tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8025/files#diff-ff84b903292a1f76994704e04dc9a7aaf6788497f8b561b7f693f6146a93b140">+199/-0</a>&nbsp; </td>

</tr>

<tr>
  <td><strong>AddUser.spec.js</strong><dd><code>Update and fix AddUser tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8025/files#diff-9f6b815996dc8497b76604e29e7495042466714655abc8b5010430959007f3d0">+62/-1</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>ReportList.spec.js</strong><dd><code>Fix failing report list tests</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8025/files#diff-c3de3f1590ac7b4401cf5570a52f0dcf6471898e4ffc7ccbc19166ba600a181f">+46/-1</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td><strong>CreateReport.spec.js</strong><dd><code>Fix create report test failures</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8025/files#diff-8a0f39751d08a258c4997c5e68edaf167ea01477528beee2295dd9f992310cc6">+61/-0</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr><tr><td><strong>Enhancement</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td><strong>handlers.ts</strong><dd><code>Update MSW handlers for functions and alerts</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></td>
  <td><a href="https://github.com/openobserve/openobserve/pull/8025/files#diff-fa72375a2848d474b311c07cf6aef7d6117b1f4d88f56a74c9207f39df72333c">+13/-3</a>&nbsp; &nbsp; </td>

</tr>
</table></details></td></tr></tr></tbody></table>

</details>

___

### CODE REVIEW

File: web/src/services/auth.spec.ts [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        }
+      };
+
+      mockHttp.mockReturnValue({
+        post: vi.fn().mockResolvedValue(mockResponse)
+      } as any);
+
+      const result = await auth.sign_in_user(loginData);
+
+      expect(mockHttp().post).toHaveBeenCalledWith("/auth/login", loginData);
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/services/auth.spec.ts:52]: it is not needed because we are checking with the params

File: web/src/services/billings.spec.ts [hunk 2]
```diff
@@ -0,0 +1,547 @@
...
+
+      expect(mockHttp().get).toHaveBeenCalledWith(
+        `/api/${mockOrgId}/billings/hosted_subscription_url?plan=${planName}`
+      );
+      expect(result).toEqual(mockResponse);
+    });
+
+    it("should handle special characters in plan name", async () => {
+      const planName = "pro plan with spaces";
+      const encodedPlanName = "pro plan with spaces";
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/services/billings.spec.ts:231]: we are checking the actual value directly

File: web/src/services/users.spec.ts [hunk 3]
```diff
@@ -0,0 +1,425 @@
...
+
+      mockHttp.mockReturnValue({
+        get: vi.fn().mockResolvedValue(mockGroups)
+      } as any);
+
+      const result = await users.getUserGroups(mockOrgId, mockUserEmail);
+
+      expect(mockHttp().get).toHaveBeenCalledWith(
+        `api/${mockOrgId}/users/${mockUserEmail}/groups`
+      );
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/services/users.spec.ts:309]: No , we dont get leading slash

File: web/src/services/users.spec.ts [hunk 4]
```diff
@@ -0,0 +1,425 @@
...
+
+      mockHttp.mockReturnValue({
+        get: vi.fn().mockResolvedValue(mockUserRoles)
+      } as any);
+
+      const result = await users.getUserRoles(mockOrgId, mockUserEmail);
+
+      expect(mockHttp().get).toHaveBeenCalledWith(
+        `api/${mockOrgId}/users/${mockUserEmail}/roles`
+      );
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/services/users.spec.ts:328]: no we dont get the leading slash

File: web/src/services/tickets.spec.ts [hunk 5]
```diff
@@ -0,0 +1,565 @@
...
+  });
+
+  describe("create", () => {
+    it("should create a new ticket", async () => {
+      const ticketData = {
+        title: "New Support Request",
+        description: "Need help with configuration",
+        priority: "medium",
+        category: "support",
+        reporter_email: "user@example.com"
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/services/tickets.spec.ts]: Thanks considered it

File: web/src/services/tickets.spec.ts [hunk 6]
```diff
@@ -0,0 +1,565 @@
...
+
+      // List tickets to verify creation
+      const ticketsList = await tickets.list(1, 10, "created_at", false, "integration");
+      expect(ticketsList.data.list).toHaveLength(1);
+      expect(ticketsList.data.list[0].title).toBe(ticketData.title);
+
+      // Update ticket
+      const updatedTicket = await tickets.update(createdTicketId, updateData);
+      expect(updatedTicket.data.status).toBe("in_progress");
+      expect(updatedTicket.data.assignee).toBe("developer@example.com");
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/services/tickets.spec.ts:487]: this is considered

File: web/src/services/tickets.spec.ts [hunk 7]
```diff
@@ -0,0 +1,565 @@
...
+      const ticketData = {
+        title: "Integration Test Ticket",
+        description: "Testing complete workflow",
+        priority: "high"
+      };
+
+      const createdTicketId = 999;
+      const updateData = {
+        status: "in_progress",
+        assignee: "developer@example.com"
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/services/tickets.spec.ts:453]: done considered it

File: web/src/services/tickets.spec.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/services/tickets.spec.ts:487]: done considered it

File: web/src/utils/common.spec.ts [hunk 8]
```diff
@@ -0,0 +1,395 @@
...
+          name: "Different User Org"
+        }
+      };
+
+      mockOrganizationsService.os_list = vi.fn().mockResolvedValue(mockOrgsResponse);
+      mockZincutils.useLocalOrganization = vi.fn().mockReturnValue(localOrg);
+
+      await getDefaultOrganization(mockUserInfo, mockOrgIdentifier);
+
+      expect(mockZincutils.useLocalOrganization).toHaveBeenCalledWith("");
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/utils/common.spec.ts:131]: this test is specifically for testing the function call

File: web/src/utils/common.spec.ts [hunk 9]
```diff
@@ -0,0 +1,395 @@
...
+      expect(logsErrorMessage(99999)).toBe("");
+      expect(logsErrorMessage(0)).toBe("");
+      expect(logsErrorMessage(-1)).toBe("");
+    });
+
+    it("should handle edge case error codes", () => {
+      // Test some edge cases that might exist in the actual implementation
+      expect(logsErrorMessage(10001)).toBe("message.ServerInternalError");
+      expect(logsErrorMessage(20001)).toBe("message.SearchSQLNotValid");
+    });
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/utils/common.spec.ts]: Thanks for identifying we will remove

File: web/src/utils/common.spec.ts [hunk 10]
```diff
@@ -0,0 +1,395 @@
...
+      expect(() => {
+        redirectUser("/test-path");
+      }).toThrow();
+
+      // Restore window.location
+      Object.defineProperty(window, 'location', {
+        value: originalLocation,
+        writable: true
+      });
+    });
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/utils/common.spec.ts]: Thanks we have covered all the error related scenarios so this was removed

File: web/src/views/LogStream.spec.ts [hunk 11]
```diff
@@ -13,113 +13,1228 @@
...
+          doc_num: 100,
+          storage_size: "50 MB",
+          compressed_size: "25 MB",
+          index_size: "10 MB",
+          storage_type: "local",
+          actions: "action buttons",
+          schema: [],
+        }
+      ];
+    });
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/views/LogStream.spec.ts:411]: we have tried to mock the exact component logStream values

File: web/src/views/LogStream.spec.ts [hunk 12]
```diff
@@ -13,113 +13,1228 @@
...
+      const invalidStream = { name: "invalid-stream!" };
+      
+      // Test stream name validation if such method exists
+      if (typeof wrapper.vm.validateStreamName === 'function') {
+        expect(wrapper.vm.validateStreamName(validStream.name)).toBe(true);
+        expect(wrapper.vm.validateStreamName(invalidStream.name)).toBe(false);
+      } else {
+        expect(validStream.name).toMatch(/^[a-zA-Z0-9_]+$/);
+        expect(invalidStream.name).not.toMatch(/^[a-zA-Z0-9_]+$/);
+      }
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/views/LogStream.spec.ts]: Thanks removed the test case

File: web/src/views/LogStream.spec.ts [hunk 13]
```diff
@@ -13,113 +13,1228 @@
...
+      if (typeof wrapper.vm.sortBy === 'function') {
+        await wrapper.vm.sortBy('name', true);
+        expect(wrapper.vm.logStream[0].name).toBe('a_stream');
+      }
+      
+      // Test document count sorting
+      if (typeof wrapper.vm.sortBy === 'function') {
+        await wrapper.vm.sortBy('doc_num', false);
+        expect(wrapper.vm.logStream[0].doc_num).toBe(200);
+      }
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/views/LogStream.spec.ts]: Thanks removed the test case which doesn't test actual function

File: web/src/views/LogStream.spec.ts [hunk 14]
```diff
@@ -13,113 +13,1228 @@
...
+      // The component should at least store the input without crashing
+      expect(wrapper.vm.filterQuery).toBe(maliciousInput);
+      
+      // Test that component doesn't crash with malicious input
+      expect(wrapper.exists()).toBe(true);
+      
+      // Reset to safe value
+      wrapper.vm.filterQuery = 'safe_query';
+      expect(wrapper.vm.filterQuery).toBe('safe_query');
+    });
```
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/views/LogStream.spec.ts:1101]: No we are checking if user inputs have been safe or not with safe_query  as input

File: web/src/services/tickets.spec.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (nikhilsaikethe) [reply web/src/services/tickets.spec.ts:487]: no we are getting without string literal
